packages/gator-permissions-snap/src/permissions/erc20TokenPeriodic/context.ts

```diff
@@ -50,7 +50,18 @@ export async function applyContext({
     permissionDetails,
     tokenMetadata: { decimals },
   } = context;
-  const expiry = convertReadableDateToTimestamp(context.expiry);
+
+  const expiry = convertReadableDateToTimestamp(context.expiry.timestamp);
+
+  const rules = originalRequest.rules?.map((rule) => {
```

Comment:
This will silently discard the expiry that the user has been shown, if there are no rules on `originalRequest` - I know other components validate that there _are_ rules, and that there is an expiry rule, but the context should probably throw if it cannot apply the expiry.

It could create an expiry rule if the original rule is not found, but that might just hide other issues.